### ISSUE TITLE
onnx mask encoder doesn't work on iOS(swift) and C++

### ISSUE THREAD

**Issue by tempdeltavalue** [facebookresearch/segment-anything#325]
Hi. 

In both cases I get strange masks which are differs from python version

iOS gist:
https://gist.github.com/tempdeltavalue/aec9c3e7d9ba15433dc427c29c45c496

result image 
<img width="627" alt="Screenshot 2023-05-01 at 23 48 31" src="https://user-images.githubusercontent.com/36921178/236181552-ab99eab0-5903-44b3-acee-c24dd53cbc4d.png">

c++ gist:
https://gist.github.com/tempdeltavalue/2ddf7d3195f336d2a1dd7d5e71e28224

Sorry for unclear variables names )

**Comment by bcojan** [comment]
Hello @tempdeltavalue in your gist above you are creating a data of size 16777216 for a dimension of `[1, 256, 64, 64]` (`let data = NSMutableData(data: Data(count: 16777216))`) - does this mean that we expect the data we pass to the onnx runtime to have each float value over 16 bits? 
in swift, when using the type Float64, it is encoded over 8 bits. 

I am also trying to run the SAM model on iOS
I have a sample app where I am calling the inference with the same input data, and while it is working fine running a python script, the call always crashes on iOS with no useful call stack. I wonder of that data I use to build the ORTValue tensorData is incorrect.

**Comment by tempdeltavalue** [comment]
@bcojan Hi. Thank you for your comment. 

Yes , it means 16 bits per float , I tried to divide it by 2 (8 bits for float) but it doesn't on anything. Also, I'm not sure what this is swift's float, because I'm using ort float type [here](https://github.com/tempdeltavalue/onnx_segment_anything_experiments/blob/main/ios/ORTOD/ORTObjectDetection/ModelHandler.swift#LL130C70-L130C70)

Do you use only mask_encoder onnx file?
If you want to you can try to run [this modified ORT demo](https://github.com/tempdeltavalue/onnx_segment_anything_experiments/tree/main/ios/ORTOD), I don't experience any crash here..

If you find something that doesn't work, any advice on how to fix it would be much appreciated.

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by luma-ai-team** [comment]
Hey @bcojan, did you solve by chance?

**Comment by anthonydito** [comment]
I have also ran into this issue. And while the mac example does work, it uses OpenCV. I am trying to find an example using simply ONNX data types. 

Has anybody been able to get that running?
